benches/parser.rs

```diff
@@ -0,0 +1,60 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+use std::{fs, path::PathBuf, str::FromStr};
+
+fn benchmark_quil_corpus(c: &mut Criterion) {
+    from_corpus().iter().for_each(|cfg| {
+        c.bench_function(&cfg.name, |b| {
+            b.iter(|| {
+                let _ = quil_rs::Program::from_str(&cfg.program);
```

Comment:
Oh! Probably want to use `criterion::black_box` as well, no?

Reply:
Good question -- in this case, I don't think there is any optimization (I have not checked the optimized IR, though!). 

I initially used `black_box`, and found that it actually made the average _faster_. I assume this is because Rust no longer needs to `drop` the value, which isn't really a real-world characteristic of a program that would use `quil-rust`. 

The benchmarks have also been run with the `Result` unwrapped, an actual variable binding, as well as an additional use of `black_box` just to use the unwrapped `Program` from the `Result`. They didn't have much impact, so I figured I'd use the simplest bit in the benchmark. Please feel free to tinker though, or I can try other suggestions - definitely don't want these results to be misleading 👍